### PR TITLE
chore: Move DTOs to `sdk-common-jvm` shared lib

### PR DESCRIPTION
**Pull request by felipecsl**
# Description

Add dependency on shared library `sdk-common-jvm`, which will power this and the Eppo java server SDK https://github.com/Eppo-exp/sdk-common-jdk/pull/1

All unit tests are passing so this is more or less ready for review already. Once we upload the `sdk-common-jvm` snapshot to sonatype, then CI builds should pass

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -9,17 +9,22 @@ on:
...
+
 jobs:
   test-android-sdk:
     runs-on: ubuntu-latest
     steps:
       - name: Check out Java SDK
         uses: actions/checkout@v4
 
-      - name: Set up JDK 11
+      - name: Set up JDK 17
```
Comment (aarsilv) [inline .github/workflows/test.yaml:24]: My understanding was we need to use Java 11 for Android. Has this changed? 🤞

**Review by aarsilv — APPROVED** [review]
Approving using the common DTOs although looks like we have a test failure--may even be unrelated but just from updating the shared test data.

File: .github/workflows/test.yaml (hunk 1 above)
Comment (felipecsl) [reply .github/workflows/test.yaml:24]: IIRC Android Gradle Plugin v8 and above require java 17 to run

**Comment by felipecsl** [comment]
> Approving using the common DTOs although looks like we have a test failure--may even be unrelated but just from updating the shared test data.

That's correct, the latest test data exposed a bug (unrelated to these changes though!) working on that now

**Comment by felipecsl** [comment]
Planning to merge this after next green build

**Pull request merged**